docs/content/guides/dagster/recommended_project_structure.mdx

```diff
@@ -0,0 +1,201 @@
+
+In complex projects, we find it helpful to make resources reusable and configured with pre-defined values via <PyObject object="configured" />. This allows your teammates to use a pre-defined resource set or make changes to shared resources, thus enabling more efficient project development.
+
+This pattern also helps you to easily execute jobs in different environments without code changes. In this example, we dynamically define a code location based on the deployment in [`__init__.py`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/\__init\_\_.py) and are able to keep all code the same across testing, local development, staging, and production. Read more about our recommendations in [Transitioning Data Pipelines from Development to Production](/guides/dagster/transitioning-data-pipelines-from-development-to-production)
+
+---
+
+## For jobs
+
+When using assets based data pipelines, we recommend having a jobs.py file which imports the assets, partitions, sensors, etc. to build each of the jobs.
+
+### When using ops and graphs
+
+This project does not include ops or graphs, if it did, this would be the recommendation on how to structure it.
+
+In these situations, we recommend having a jobs folder rather than a jobs.py file. Depending on the types of jobs you have, you can create a seperate file for each types of jobs.
+
+The ops and graphs would be defined in the job file with the job definition which utlizes the ops/graph in the file too.
+
+```shell
+├──project_with_ops
+    ...
+│   ├── jobs
+│   │   ├── jobs_using_assets.py
+│   │   ├── jobs_using_ops_assets.py
+│   │   ├── jobs_using_ops.py
+│   │   ├── jobs_using_ops_graphs.py
+```
+
+---
+
+## For multiple code locations
+
+So far, we've discussed our recommendations for structuring a large project which contains only one code location. Dagster also allows you to structure a project with multiple definitions. We don't recommend over-abstracting too early, and in most cases, code location should be sufficient. The pattern we found useful is to use multiple code location is to keep conflicting dependencies separate, where each definition can keep their own package requirements (e.g., `setup.py`) and deployment specs (e.g., `Dockerfile`).
+
+---
+
+## For tests
+
+We recommend setting up a separate test folder structure that mirrors the main project, (e.g. having a folder for test assets with any applicable subfolders) which contains the unit tests for each of the components of the data pipeline.
+
+- Each of the components in Dagster such as assets, sensors and resources can all be tested separately. See more on [Testing in Dagster](https://docs.dagster.io/concepts/testing)
+- For details on [Unit Tests with stubs and mocks](/guides/dagster/transitioning-data-pipelines-from-development-to-production#advanced-unit-tests-with-stubs-and-mocks)
```

Comment:
❓ This doesn't seem like a complete thought - is there supposed to be more here?

Reply:
removing